### PR TITLE
Normal formatting & fixes generate_wrappers.sh

### LINKED ISSUE #378: Fix generate_wrappers script
Per https://github.com/servo/mozjs/pull/377#issuecomment-1693493186 it looks like it's not working correctly.

### PR DESCRIPTION
**Pull request by sagudev**
Fixes #378 

- normal fmt everywhere (gen_wrapper does own formatting on cloned files)
- gen_wrapper script repaired and now works with both gsed and sed
- wrapper files regenerated

### CODE REVIEW

File: rust-mozjs/src/rust.rs [hunk 1]
```diff
@@ -1135,6 +1135,7 @@ pub mod wrappers {
     use jsapi::{JSStructuredCloneCallbacks, JSStructuredCloneReader, JSStructuredCloneWriter};
     use jsapi::{MallocSizeOf, ObjectOpResult, ObjectPrivateVisitor, TabSizes};
     use jsapi::{SavedFrameResult, SavedFrameSelfHosted};
+    use mozjs_sys::jsapi::JS::HandleObject;
```
Comment (sagudev) [inline rust-mozjs/src/rust.rs:1138]: This change might be problematic. Will need to review public surface for this.

Comment (sagudev) [reply rust-mozjs/src/rust.rs]: This is now properly solved

File: rust-mozjs/src/jsapi_wrappers.in [hunk 2]
```diff
@@ -38,7 +38,7 @@ wrap!(jsapi: pub fn ArrayBufferCopyData(cx: *mut JSContext, toBlock: HandleObjec
 wrap!(jsapi: pub fn ArrayBufferClone(cx: *mut JSContext, srcBuffer: HandleObject, srcByteOffset: usize, srcLength: usize) -> *mut JSObject);
 wrap!(jsapi: pub fn ComputeThis(cx: *mut JSContext, vp: *mut Value, thisObject: MutableHandleObject) -> bool);
 wrap!(jsapi: pub fn MaybeFreezeCtorAndPrototype(cx: *mut JSContext, ctor: HandleObject, maybeProto: HandleObject) -> bool);
-wrap!(jsapi: pub fn GetRealmObjectPrototypeHandle(cx: *mut JSContext) -> HandleObject);
+wrap!(jsapi: pub fn GetRealmObjectPrototypeHandle(cx: *mut JSContext) -> mozjs_sys::jsapi::JS::HandleObject);
```
Comment (sagudev) [inline rust-mozjs/src/jsapi_wrappers.in:41]: Actually this needs to be wrapped or removed.

File: rust-mozjs/src/jsapi_wrappers.in [hunk 3]
```diff
@@ -146,7 +146,7 @@ wrap!(jsapi: pub fn GetWasmModule(obj: HandleObject) -> RefPtr<WasmModule>);
 wrap!(jsapi: pub fn Call(cx: *mut JSContext, thisv: Handle<Value>, fun: Handle<Value>, args: *const HandleValueArray, rval: MutableHandle<Value>) -> bool);
 wrap!(jsapi: pub fn Construct(cx: *mut JSContext, fun: Handle<Value>, newTarget: HandleObject, args: *const HandleValueArray, objp: MutableHandleObject) -> bool);
 wrap!(jsapi: pub fn Construct1(cx: *mut JSContext, fun: Handle<Value>, args: *const HandleValueArray, objp: MutableHandleObject) -> bool);
-// wrap!(jsapi: pub fn GetDebuggeeGlobals(cx: *mut JSContext, dbgObj: *mut JSObject, vector: MutableHandleObjectVector) -> bool);
+wrap!(jsapi: pub fn GetDebuggeeGlobals(cx: *mut JSContext, dbgObj: *mut JSObject, vector: MutableHandleObjectVector) -> bool);
```
Comment (sagudev) [inline rust-mozjs/src/jsapi_wrappers.in:149]: this one is also redundant

File: rust-mozjs/src/jsapi_wrappers.in (hunk 2 above)
Comment (sagudev) [reply rust-mozjs/src/jsapi_wrappers.in]: this one cannot be wrapped

File: rust-mozjs/src/glue/rustfmt.toml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Try to get all function signatures to fit on one line, so the jsapi file can be post-processed more easily
+max_width = 500
```
Comment (mukilan) [inline rust-mozjs/src/glue/rustfmt.toml:2]: I'm wondering if glue.rs should be easy to read - would it be possible to keep the default line length limit and modify the generate_wrappers.sh script instead? Is it only the cases where we have `pub fn method(\n..` that needs to be handled to make this work or is it more complicated?

**Comment by mukilan** [comment]
Thank you for this fix! Overall it looks good, but I've left a comment about the change in formatting of glue.rs. Also, I'm looking into the windows build failure (I'm trying to switch to windows-2019 for now).

**Comment by sagudev** [comment]
>  Also, I'm looking into the windows build failure (I'm trying to switch to windows-2019 for now).

Actually as noted in #368 the problem is https://github.com/actions/runner-images/issues/8125 so manually installing llvm 16 should solve the problem. But I would just wait for the fix to be deployed it should be pretty soon as there was no deployment on windows for nearly two weeks.

File: rust-mozjs/src/glue/rustfmt.toml (hunk 4 above)
Comment (sagudev) [reply rust-mozjs/src/glue/rustfmt.toml]: This works even better

File: rust-mozjs/src/glue/rustfmt.toml
Comment (mukilan) [inline rust-mozjs/src/glue/rustfmt.toml:1]: Do we need this rustfmt.toml file still?

Comment (sagudev) [reply rust-mozjs/src/glue/rustfmt.toml:1]: No we do not.

**Review by mukilan — APPROVED** [review]
Thank you!

**Pull request merged**